File: backend/app/models/generation.py
"""
API request/response models for the generation pipeline.
"""
from pydantic import BaseModel
from typing import Optional, List, Dict


class TaskModelConfig(BaseModel):
    """Per-task model configuration"""
    provider: str = "gemini"
    model: str = "gemini-2.5-pro"


class LLMConfig(BaseModel):
    """LLM provider configuration — supports per-task model routing."""
    default_provider: str = "gemini"
    default_model: str = "gemini-2.5-pro"
    tasks: Dict[str, TaskModelConfig] = {}


class GenerationRequest(BaseModel):
    application_id: str
    job_title: str
    company_name: str
    company_description: Optional[str] = None
    job_description: str
    selected_project_ids: List[str]
    tone: str = "professional"


class MatchRequest(BaseModel):
    job_title: str
    company_name: str
    job_description: str


class MatchResult(BaseModel):
    project_id: str
    project_name: str
    relevance_score: float
    reasoning: str


class PointsRegenerateRequest(BaseModel):
    application_id: str
    section_key: str
    custom_instructions: Optional[str] = None


class ResumeExportRequest(BaseModel):
    application_id: str


class SSEEvent(BaseModel):
    """Server-Sent Event payload for streaming generation progress."""
    event: str  # "stage" | "token" | "section_complete" | "error" | "complete"
    data: dict

File: frontend/src/types/generation.ts
/**
 * TypeScript interfaces mirroring backend/app/models/generation.py
 */

export interface TaskModelConfig {
  provider: string;
  model: string;
}

export interface LLMConfig {
  default_provider: string;
  default_model: string;
  tasks: Record<string, TaskModelConfig>;
}

export interface GenerationRequest {
  application_id: string;
  job_title: string;
  company_name: string;
  company_description?: string;
  job_description: string;
  selected_project_ids: string[];
  tone: string;
}

export interface MatchRequest {
  job_title: string;
  company_name: string;
  job_description: string;
}

export interface MatchResult {
  project_id: string;
  project_name: string;
  relevance_score: number;
  reasoning: string;
}

export interface PointsRegenerateRequest {
  application_id: string;
  section_key: string;
  custom_instructions?: string;
}

export interface ResumeExportRequest {
  application_id: string;
}

export interface SSEEvent {
  event: 'stage' | 'token' | 'section_complete' | 'error' | 'complete';
  data: Record<string, unknown>;
}
